astropy/convolution/convolve.py

```diff
@@ -195,7 +195,7 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
         array_internal[mask != 0] = np.nan
     if np.ma.is_masked(kernel):
         # *kernel* doesn't support NaN interpolation, so instead we just fill it
-        kernel_internal = kernel.filled(fill_value)
+        kernel_internal = kernel_internal.filled(fill_value)
```

Comment:
If ``kernel`` is __not__ a float, it's previous conversion to a ``float`` array ([here](https://github.com/astropy/astropy/pull/7542/files#diff-f632180d9f1e3fcf689f62e4c424e94fR175)) gets undone as ``kernel.filled()`` returns a copy of the original ``kernel`` type.